integration-tests/testreporters/keeper_benchmark.go

```diff
@@ -0,0 +1,299 @@
+		fmt.Sprintf("%.2f%%", pctWithinSLA),
+		fmt.Sprintf("%.2f%%", pctReverted),
+	})
+	if err != nil {
+		return err
+	}
+	keeperReportWriter.Flush()
+	log.Info().
+		Int64("Total Times Eligible", totalEligibleCount).
+		Int64("Total Performed", totalPerformed).
+		Int64("Total Reverted", totalReverted).
+		Float64("Average Perform Delay", avg).
+		Int64("Median Perform Delay", median).
+		Int64("90th pct Perform Delay", ninetyPct).
+		Int64("99th pct Perform Delay", ninetyNinePct).
+		Int64("Max Perform Delay", max).
+		Float64("Percent Within SLA", pctWithinSLA).
+		Float64("Percent Reverted", pctReverted).
+		Msg("Calculated Aggregate Results")
+
+	err = keeperReportWriter.Write([]string{
+		"Contract Index",
+		"Contract Address",
+		"Total Times Eligible",
+		"Total Performed Upkeeps",
+		"Average Perform Delay",
+		"Median Perform Delay",
+		"90th pct Perform Delay",
+		"99th pct Perform Delay",
+		"Largest Perform Delay",
+		"Percent Within SLA",
+	})
+	if err != nil {
+		return err
+	}
+
+	for contractIndex, report := range k.Reports {
+		avg, median, ninetyPct, ninetyNinePct, max := intListStats(report.AllCheckDelays)
+		err = keeperReportWriter.Write([]string{
+			fmt.Sprint(contractIndex),
+			report.ContractAddress,
+			fmt.Sprint(report.TotalEligibleCount),
+			fmt.Sprint(report.TotalPerformedUpkeeps),
+			fmt.Sprintf("%.2f", avg),
+			fmt.Sprint(median),
+			fmt.Sprint(ninetyPct),
+			fmt.Sprint(ninetyNinePct),
+			fmt.Sprint(max),
+			fmt.Sprintf("%.2f%%", (1.0-float64(report.TotalSLAMissedUpkeeps)/float64(report.TotalEligibleCount))*100),
+		})
+		if err != nil {
+			return err
+		}
+	}
+	keeperReportWriter.Flush()
+
+	txs, err := json.Marshal(k.AttemptedChainlinkTransactions)
+	if err != nil {
+		return err
+	}
+	err = os.WriteFile(k.attemptedTransactionsFile, txs, 0600)
+	if err != nil {
+		return err
+	}
+
+	log.Info().Msg("Successfully wrote report on Keeper Benchmark")
+
+	k.Summary.Metrics.Delay = map[string]interface{}{
+		"mean":   avg,
+		"median": median,
+		"90p":    ninetyNinePct,
+		"99p":    ninetyNinePct,
+		"max":    max,
+	}
+	k.Summary.Metrics.PercentWithinSLA = pctWithinSLA
+	k.Summary.Metrics.PercentRevert = pctReverted
+	k.Summary.Metrics.TotalTimesEligible = totalEligibleCount
+	k.Summary.Metrics.TotalTimesPerformed = totalPerformed
+	k.Summary.Metrics.AverageActualPerformsPerBlock = float64(totalPerformed) / float64(k.Summary.TestInputs["BlockRange"].(int64))
+
+	// TODO: Set test expectations
+	/* Expect(int64(pctWithinSLA)).Should(BeNumerically(">=", int64(80)), "Expected PercentWithinSLA to be greater than or equal to 80, but got %f", pctWithinSLA)
+	Expect(int64(pctReverted)).Should(BeNumerically("<=", int64(10)), "Expected PercentRevert to be less than or equal to 10, but got %f", pctReverted)
+	Expect(k.Summary.Metrics.AverageActualPerformsPerBlock).Should(BeNumerically("~", k.Summary.Load.AverageExpectedPerformsPerBlock, 10), "Expected PercentRevert to be less than 10, but got %f", pctReverted) */
```

Comment:
nit. I am not sure if we should fail the testRunner if results aren't good. Maybe you can clean it up?